### ISSUE TITLE
contextmenu event should dissociate right click event from rotate/tilt event

### ISSUE THREAD

**Issue by Dahkon** [mapbox/mapbox-gl-js#8824]
I'm trying to implement a right click (long tap on mobile device) event that add a marker unfortunatly, when we rotate / tilt map with the right click button the marker is added too.

```
map.on('contextmenu', function(e) {
	console.log(e);
       /* adding marker is not a good idea here */
});
```

Expected behavior : 
right click / long tap : add the marker
right click move (tilt / rotate) : don't add the marker

I think states could be added to the "map mouse event" to indicate if tilt or rotate occures when contextmenu fired or maybe some other events ? (long tap, right click for exemple)

Tests are done on chrome desktop for now.

I'm also open to any workaround. Thanks,

**Comment by Dahkon** [comment]
Here you go, still occurs with v1.4.0 : https://jsfiddle.net/z92tnvdp/

Test drive : 
Right click on the map => add a marker
Right click to rotate, tilt => add a marker but shouldn't

**Comment by kkaefer** [comment]
I tried this in Firefox and Chrome and couldn't reproduce the behavior you're describing. A regular right click adds the marker, while right-click + pan tilts and rotates the map, but does _not_ add a marker.

**Comment by Dahkon** [comment]
Did you try on the jsfiddle ?
It's still occuring on Chrome different PC though AND firefox (Windows) with a regular 2 buttons mouse.

The problem is the same on our app or on the jsfiddle.
When you hold right mouse button to tilt / rotate, on mouse up the marker is added BUT shouldn't.

**Comment by kkaefer** [comment]
@Dahkon yes, I used your JSFiddle. What browser versions are you using?

**Comment by kkaefer** [comment]
> When you hold right mouse button to tilt / rotate, on mouse up the marker is added BUT shouldn't.

That is the behavior I'm observing. Once the tilt/rotatehandler kicks in, no icon is added.

**Comment by Dahkon** [comment]
> @Dahkon yes, I used your JSFiddle. What browser versions are you using?

Browsers are UP to date but I tried on severals pc (not on mac). Same behavior.

**Comment by kkaefer** [comment]
> on severals pc (not on mac)

that was the missing piece. I assume with PC you mean Windows. I just tried your JSFiddle on Windows 10 in Chrome 78, Firefox 70, Edge 44, and IE 11 and could reproduce this behavior both with a trackpad and with an external mouse.

**Comment by Dahkon** [comment]
> > on severals pc (not on mac)
> 
> that was the missing piece. I assume with PC you mean Windows. I just tried your JSFiddle on Windows 10 in Chrome 78, Firefox 70, Edge 44, and IE 11 and could reproduce this behavior both with a trackpad and with an external mouse.

Yes Windows 10, 8.1 and 7 (not tested under). Sorry I should have said that earlier.
